src/chord.js

```diff
@@ -1,368 +1,108 @@
-    if (this.getCurrentChord(this.getChordNameAttr())) {
-      this.initAttributes();
-
-      this.findBorders();
-      this.initSize();
-      this.shadowRoot.innerHTML = this.getCurrentTemplete();
-      this.initCanvas();
-
-      this.calculateCanvasSize();
-      this.canvasSettings.canvas.setAttribute(
-        "height",
-        this.canvasSettings.canvasHeight
-      );
-      this.canvasSettings.canvas.setAttribute(
-        "width",
-        this.canvasSettings.canvasWidth
-      );
-      this.drawChord();
-    } else {
-      this.shadowRoot.innerHTML = "<div>Unknown chord</div>";
-    }
-  }
 
-  initSize() {
-    this.canvasSettings.stringHeight =
-      this.canvasSettings.stringHeight * this.customAttributes.size;
-    this.canvasSettings.rowWidth =
-      this.canvasSettings.stringHeight * this.customAttributes.size;
-  }
-
-  getChordNameAttr() {
-    return this.chordComponentRef.getAttribute("chord");
-  }
-
-  getSizeAttr() {
-    return this.chordComponentRef.getAttribute("size");
-  }
-
-  getThemeAttr() {
-    return this.chordComponentRef.getAttribute("theme");
-  }
+    this.attrGetter = new AttrGetter(this.chordComponentRef);
 
-  getReflectAttr() {
-    return this.chordComponentRef.getAttribute("reflect");
-  }
+    if (this.attrGetter.chordArray.length) {
+      this.initAttributes();
 
-  resolveReflectAttr(attrStr) {
-    const reflectModel = { horizontal: false, vertical: false };
-    const reflectData =
-      attrStr && attrStr.length ? attrStr.toLowerCase().split(/\s+|[,.]+/) : "";
-    if (reflectData) {
-      reflectData.map(el => {
-        if (el == "horizontal" || el == "x") {
-          reflectModel.horizontal = true;
-        }
-        if (el == "vertical" || el == "y") {
-          reflectModel.vertical = true;
-        }
+      this.shadowRoot.innerHTML = this.resolveFinalTemplate();
+      this.customAttributes.chords.map(chord => {
+        const customAttributes = {
+          ...this.customAttributes,
+          reflection: this.canvasSettings.reflect
+        };
+        new CanvasDrawTool(
+          this.initCanvas(chord),
+          chord,
+          customAttributes
+        ).draw();
       });
+    } else {
+      this.shadowRoot.innerHTML =
+        this.templatesHolder.unknownChordTemplate +
+        stylesHolder.getUnknownChordTemplateStyleTag();
     }
-    this.canvasSettings.reflect = reflectModel;
   }
 
   initAttributes() {
-    this.resolveSize(this.getSizeAttr());
-
-    this.customAttributes.currentChord = this.getCurrentChord(
-      this.getChordNameAttr()
-    );
-    this.resolveTheme(this.getThemeAttr());
-
-    this.resolveReflectAttr(this.getReflectAttr());
+    this.initChords();
+    this.initSize();
+    this.initTheme();
+    this.initReflectAttr();
   }
 
-  resolveTheme(themeStr) {
-    this.customAttributes.theme =
-      themeStr && themeStr.toLowerCase() == "dark" ? "dark" : "light";
+  initSize() {
+    this.customAttributes.size = this.attrGetter.size;
```

Comment:
shouldn't we handle here case when attrGetter.size is null and aa a default value?

Reply:
it is handled in attributeGetter